### PR TITLE
doc: Updated the OS recommendations for newer Ceph releases

### PR DESCRIPTION
**Pull request by ritz303**
Fixes: 12784

Signed-off-by: ritz303 <ritz_303@yahoo.com>

### CODE REVIEW

File: doc/start/os-recommendations.rst [hunk 1]
```diff
@@ -15,36 +15,22 @@ Linux Kernel
...
+  - 4.1.4 or later
+  - 3.16.3 or later (rbd deadlock regression in 3.16.[0-2])
+  - *NOT* 3.15.* (rbd deadlock regression)
+  - 3.14.*
 
   These are considered pretty old, but if you must:
 
-  - v3.6.6 or later in the v3.6 stable series
-  - v3.4.20 or later in the v3.4 stable series
+  - 3.10.* or earlier
```
Comment (idryomov) [inline doc/start/os-recommendations.rst]: Drop "or earlier".  Otherwise the kernel client part LGTM.

File: doc/start/os-recommendations.rst [hunk 2]
```diff
@@ -54,79 +40,69 @@ platforms.  Generally speaking, there is very little dependence on
...
 +----------+----------+--------------------+--------------+---------+------------+
 
-Argonaut (0.48), Bobtail (0.56), and Cuttlefish (0.61)
-------------------------------------------------------
-
-The Ceph Argonaut, Bobtail, and Cuttlefish releases are no longer supported,
-and users should update to the latest stable release (Dumpling or Firefly).
-
+**NOTE**: Ceph also supports ``Quantal``, ``Raring`` and ``Saucy``. However, we 
+recommend using LTS releases.
```
Comment (liewegas) [inline doc/start/os-recommendations.rst]: Drop this.. non-LTS ubuntu releases not supported.

**Comment by dachary** [comment]
@ritz303 please ignore the known false negative from the bot ( http://tracker.ceph.com/issues/13554 ).

File: doc/start/os-recommendations.rst (hunk 1 above)
Comment (ritz303) [inline doc/start/os-recommendations.rst]: Thanks for the feedback!  I've removed "or earlier", see commit 03e556b.

File: doc/start/os-recommendations.rst (hunk 2 above)
Comment (ritz303) [inline doc/start/os-recommendations.rst]: Thanks for the feedback!  I've removed NOTE section, see commit 03e556b.

**Pull request merged**